pkg/repository/crypto/keys.go

```diff
@@ -0,0 +1,75 @@
+// Copyright 2020 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package crypto
+
+import (
+	"errors"
+)
+
+var (
+	// ErrorKeyUninitialized will be present when key is used before Deserialize called
+	ErrorKeyUninitialized = errors.New("Key not initialized, call Deserialize first")
+	// ErrorDeserializeKey means the key format is not valid
+	ErrorDeserializeKey = errors.New("Error on deserialize key, check if the key is valid")
+)
+
+// KeyStore is the collection of all public keys available to TiUp.
+type KeyStore interface {
+	// Add put new key into KeyStore
+	Add(string, PubKey) KeyStore
+
+	// Get return a key by it's id
+	Get(string) PubKey
+}
+
+// Serializable represents object that can be serialized and deserialized
+type Serializable interface {
+	// Translate the key to the format that can be stored
+	Serialize() ([]byte, error)
+
+	// Deserialize a key from data
+	Deserialize([]byte) error
+}
+
+// PubKey is a public key available to TiUp
+type PubKey interface {
```

Comment:
Should also add a method to get fingerprint of the key

Reply:
Why need this now?

Reply:
The key-id has the same function as this, but it should be added into KeyInfo, I think

Reply:
I believe the value of key id is fingerprint of that key, I didn't see that part, maybe I missed something?